src/main/java/seedu/address/model/lesson/RecurringLesson.java

```diff
@@ -82,4 +82,21 @@ private RecurringLesson updateRecurringLessonCount() {
                 getSubject(), getHomework());
     }
 
+    /**
+     * Returns true if this {@code RecurringLesson} clashes with the given {@code Lesson}.
+     *
+     * @param otherLesson The other lesson to be compared with.
+     * @return True if and only if lessons clash.
+     */
+    @Override
+    public boolean isClashing(Lesson otherLesson) {
+        if (otherLesson.isRecurring()) {
+            return getDayOfWeek().equals(otherLesson.getDayOfWeek()) // same day
+                    && getTimeRange().isClashing(otherLesson.getTimeRange());
+        } else {
+            return getLocalDate().compareTo(otherLesson.getLocalDate()) <= 0 // same date or before
+                    && getDayOfWeek().equals(otherLesson.getDayOfWeek()) // same day
+                    && getTimeRange().isClashing(otherLesson.getTimeRange());
```

Comment:
here too, i think we can abstract out into the parent `Lesson` class to get a boolean function that checks for all these since currently these seems quite repititive

Reply:
i see what you mean did not catch that portion